app/observers/message_observer.rb

```diff
@@ -108,4 +105,17 @@ def plan_changed_publish_event!(contract)
   def should_notify?(contract)
     NotificationCenter.new(contract).enabled?
   end
+
+  def notify_plan_change_provider(contract)
+    if contract.provider_account.provider_can_use?(:new_notification_system)
+      plan_changed_publish_event!(contract)
+    elsif should_notify?(contract)
+      contract.messenger.plan_change(contract).deliver
+    end
+  end
+
+  def notify_plan_change_developer(contract)
+    return unless contract.is_a? Cinstance
+    contract.messenger.plan_change_for_buyer(contract).deliver
```

Comment:
Might be good to use asynchronous mailing.
But I think it is OK for now as it was like that in the controller anyway.